### ISSUE TITLE
size mismatch while training

### ISSUE THREAD

**Issue by ustczhouyu** [MichalBusta/DeepTextSpotter#76]
Help, Help!
 
I add some layers in both model_cz.prototxt and tiny.prototxt, when I train the model by python train.py, one error occur: valueError: cannot reshape array of size 6204 into shape(22,1,141).This error happens in 260th line in validation.py, that is ctc_f = ctc_f.reshape(ctc_f.shape[0], ctc_f.shape[1], ctc_f.shape[3]). As we know, 6204=2*22*1*141, so I want to know than can I change ctc_f = ctc_f.reshape(ctc_f.shape[0], ctc_f.shape[1], ctc_f.shape[3]) to ctc_f = ctc_f.reshape(ctc_f.shape[0], 2*ctc_f.shape[1], ctc_f.shape[3]) so that the result is (22,2,141)?
Can somebody help me? Thank you very much.

**Comment by MichalBusta** [comment]
Hi, 
the output of OCR should reduce dim of height to one - so you have just 2D sequence at the end (N C 1 W). 
So check padding etc. on new layers ...

**Comment by ustczhouyu** [comment]
Hi Michal Busta,
I want to re-implement your model and I also want to add some attention mechanism, I have some difficulties and  I need your help, please.
1.what is the following GenericVocabulary.txt in validation.py if I want to validate icdar2013-test? It is ok if I set it to synthtext/dict_voc.txt?

  def validate(nets, dataloader, image_size = [480, 480], split_words = True):
     cmp_trie.load_dict('/home/busta/data/icdar2013-Test/GenericVocabulary.txt')

2. If I want to  replace ctc to  lstm layer+attention mechanism in the model_cz.prototxt, do I still need to change somewhere else(for example loss function ..)? 


Sincerely hope to get your reply, thank you！


At 2018-11-15 16:45:36, "Michal Busta" <notifications@github.com> wrote:


Hi,
the output of OCR should reduce dim of height to one - so you have just 2D sequence at the end (N C 1 W).
So check padding etc. on new layers ...

**Comment by ustczhouyu** [comment]
Hi Michal Busta,
I want to re-implement your model and I also want to add some attention mechanism, I have some difficulties and  I need your help, please.
1. How to pre-train the detection CNN using the SynthText dataset, pre-trained the recognition CNN on the Synthetic Word dataset? I just know that python train.py can train both detection CNN and
recognition CNN, but I don't know how to train them Individually.



2.what is the following GenericVocabulary.txt in validation.py if I want to validate icdar2013-test? It is ok if I set it to synthtext/dict_voc.txt?

  def validate(nets, dataloader, image_size = [480, 480], split_words = True):
     cmp_trie.load_dict('/home/busta/data/icdar2013-Test/GenericVocabulary.txt')

3. If I want to  replace ctc to  lstm layer+attention mechanism in the model_cz.prototxt, do I still need to change somewhere else(for example loss function ..)? 


Sincerely hope to get your reply, thank you！


At 2018-11-15 16:45:36, "Michal Busta" <notifications@github.com> wrote:


Hi,
the output of OCR should reduce dim of height to one - so you have just 2D sequence at the end (N C 1 W).
So check padding etc. on new layers ...

**Comment by ustczhouyu** [comment]
Hi Michal Busta.
I found your work very interesting and creative especially E2E-MLT - an Unconstrained End-to-End Method for Multi-Language Scene Text，I want to train my dataset, but I failed to do it because I am confused about some paths, for example when I use ICDAR2015 to train the model,

Inside the filesample_train_data/MLT/trainMLT.txt are icdar2015 localization training images such as icdar-2015-Ch4/Train/img_1.jpg and inside sample_train_data/MLT_CROPS/gt.txt are icdar2015 recognition training images such as ﻿word_1.png, "Genaxis Theatre".
I have not changed other paths. When I train the model by:
python3 train.py -train_list=sample_train_data/MLT/trainMLT.txt -batch_size=8 -num_readers=5 -debug=0 -input_size=512 -ocr_batch_size=256 -ocr_feed_list=sample_train_data/MLT_CROPS/gt.txt

the output are:
root@10ca3ad2a7d1:/home/zy/jupyter/recognition/spotter/E2E-MLT-master# python3 train.py -train_list=sample_train_data/MLT/trainMLT.txt -batch_size=8 -num_readers=5 -debug=0 -input_size=512 -ocr_batch_size=256 -ocr_feed_list=sample_train_data/MLT_CROPS/gt.txt
Using E2E-MLT
loading model from e2e-mlt.h5
e2e-mlt.h5
1000 training images in sample_train_data/MLT/trainMLT.txt
1000 training images in sample_train_data/MLT/trainMLT.txt
1000 training images in sample_train_data/MLT/trainMLT.txt
1000 training images in sample_train_data/MLT/trainMLT.txt
1000 training images in sample_train_data/MLT/trainMLT.txt
4468 training images in sample_train_data/MLT_CROPS/gt.txt
4468 training images in sample_train_data/MLT_CROPS/gt.txt
I waited for half an hour, but no more output. I want to trouble you to help me with this problem, thank you.
Yu Zhou


At 2018-11-15 16:45:36, "Michal Busta" <notifications@github.com> wrote:


Hi,
the output of OCR should reduce dim of height to one - so you have just 2D sequence at the end (N C 1 W).
So check padding etc. on new layers ...
